File: bug67_duplicate_processing_in_loop/bug67_duplicate_processing_in_loop/main1.py
def run():
    data = [0x100 * (i + 1) for i in range(10)]
    acc = 0

    for i, v in enumerate(data):
        val = v * 0x9E3779B97F4A7C15

        acc ^= val  # ✅ FIX: removed duplicate XOR
        acc &= 0xFFFFFFFFFFFFFFFF

        shift = (i % 9 + 1)
        acc = ((acc << shift) | (acc >> (64 - shift))) & 0xFFFFFFFFFFFFFFFF  # ✅ FIX

    print(hex(acc))  # ✅ Added output

run()

File: bug67_duplicate_processing_in_loop/bug67_duplicate_processing_in_loop/main1.cpp
#include <iostream>

int main() {
    int data[10];
    for (int i = 0; i < 10; i++) data[i] = 0x100 * (i + 1);
    
    unsigned long long acc = 0;

    for (int i = 0; i < 10; i++) {
        unsigned long long v = (unsigned long long)data[i] * 0x9E3779B97F4A7C15ULL;

        acc ^= v; // ✅ FIX: removed duplicate XOR

        int shift = (i % 9 + 1);
        acc = (acc << shift) | (acc >> (64 - shift)); // ✅ FIX
    }

    std::cout << std::hex << acc << std::endl; // ✅ Added output
    return 0;
}